packages/docs-app/src/examples/core-examples/cardExample.tsx

```diff
@@ -16,64 +16,47 @@
 
 import * as React from "react";
 
-import { Button, Card, Classes, type Elevation, FormGroup, H5, Slider, Switch } from "@blueprintjs/core";
-import { Example, type ExampleProps } from "@blueprintjs/docs-theme";
+import { Button, Card, Classes, Elevation, FormGroup, H5, Slider, Switch } from "@blueprintjs/core";
+import { Example, type ExampleProps, handleBooleanChange } from "@blueprintjs/docs-theme";
 
-export interface CardExampleState {
-    compact: boolean;
-    elevation: Elevation;
-    interactive: boolean;
-    selected: boolean;
-}
+const MAX_ELEVATION = 4;
 
-export class CardExample extends React.PureComponent<ExampleProps, CardExampleState> {
-    public state: CardExampleState = {
-        compact: false,
-        elevation: 0,
-        interactive: false,
-        selected: false,
-    };
+export const CardExample: React.FC<ExampleProps> = props => {
+    const [compact, setCompact] = React.useState(false);
+    const [elevation, setElevation] = React.useState<Elevation>(Elevation.ZERO);
+    const [interactive, setInteractive] = React.useState(false);
+    const [selected, setSelected] = React.useState(false);
 
-    public render() {
-        return (
-            <Example options={this.renderOptions()} {...this.props}>
-                <Card {...this.state}>
-                    <H5>Analytical applications</H5>
-                    <p>
-                        User interfaces that enable people to interact smoothly with data, ask better questions, and
-                        make better decisions.
-                    </p>
-                    <Button text="Explore products" className={Classes.BUTTON} />
-                </Card>
-            </Example>
-        );
-    }
+    const handleElevationChange = React.useCallback((value: number) => setElevation(value as Elevation), []);
 
-    private renderOptions = () => (
+    const options = (
         <>
             <H5>Props</H5>
-            <Switch checked={this.state.interactive} label="Interactive" onChange={this.handleInteractiveChange} />
-            {this.state.interactive && (
-                <Switch checked={this.state.selected} label="Selected" onChange={this.handleSelectedChange} />
-            )}
-            <Switch checked={this.state.compact} label="Compact" onChange={this.handleCompactChange} />
+            <Switch checked={interactive} label="Interactive" onChange={handleBooleanChange(setInteractive)} />
+            {interactive && <Switch checked={selected} label="Selected" onChange={handleBooleanChange(setSelected)} />}
```

Comment:
Nit: For more transparency and I think consistency with other examples, I think we should make this disabled instead of hidden when interactive is false.

Reply:
Good idea! Added in 097990c4944edb2dd3c2fe9ea4a4fd2bc4ca7c1d